compiler/rustc_trait_selection/src/solve/search_graph/mod.rs

```diff
@@ -176,17 +176,16 @@ impl<'tcx> SearchGraph<'tcx> {
             self.stack.push(StackElem { goal, has_been_used: false });
             false
         } else {
-            self.try_move_finished_goal_to_global_cache(tcx, stack_elem);
             true
         }
     }
 
     fn try_move_finished_goal_to_global_cache(
         &mut self,
         tcx: TyCtxt<'tcx>,
```

Comment:
This code is invoked outside of the `with_anon_task`, so is forbidden to invoke queries. Can we avoid giving it access to a `TyCtxt`?